app/controllers/users/reset_passwords_controller.rb

```diff
@@ -1,5 +1,7 @@
 module Users
   class ResetPasswordsController < Devise::PasswordsController
+    before_action :store_sp_metadata_in_session, only: [:edit]
```

Comment:
Another option could be to include the `request_id` parameter when redirecting the user back to the "Sign in" page below, since `request_id` is submitted as part of the form.

https://github.com/18F/identity-idp/blob/3a095866041d563e8da1fc94c4060723f0940c16/app/controllers/users/reset_passwords_controller.rb#L131

➡️  `redirect_to new_user_session_url(request_id: params[:request_id])`

I'm inclined to do it as early as possible though. For example, this implementation is also different from the email confirmation controller, which only assigns it to the session when the user submits the form after entering their password:

https://github.com/18F/identity-idp/blob/3a095866041d563e8da1fc94c4060723f0940c16/app/controllers/sign_up/passwords_controller.rb#L21